File: the-nest-backend/app/schemas/message.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class MessageCreate(BaseModel):
    content: str


class MessageResponse(BaseModel):
    id: UUID
    sender_id: UUID
    recipient_id: UUID
    content: str
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationPartner(BaseModel):
    username: str
    display_name: str
    avatar_url: Optional[str] = None


class ConversationResponse(BaseModel):
    other_user: ConversationPartner
    latest_message: str
    latest_at: datetime
    unread_count: int

File: the-nest-frontend/types/message.ts
export interface ConversationPartner {
  username: string
  display_name: string
  avatar_url: string | null
}

export interface Conversation {
  other_user: ConversationPartner
  latest_message: string
  latest_at: string
  unread_count: number
}

export interface Message {
  id: string
  sender_id: string
  recipient_id: string
  content: string
  is_read: boolean
  created_at: string
}
